huxley/api/permissions.py

```diff
@@ -222,6 +223,40 @@ def has_permission(self, request, view):
         return user_is_advisor(request, view, school_id)
 
 
+class CommitteeFeedbackListPermission(permissions.BasePermission):
+    '''Accept GET for only the chair of the committee'''
+
+    def has_permission(self, request, view):
+        user = request.user
+        if user.is_superuser:
+            return True
+
+        method = request.method
+        committee_id = request.query_params.get('committee', -1)
+        return method == 'GET' and user_is_chair(request, view, committee_id)
+
+
+class CommitteeFeedbackDetailPermission(permissions.BasePermission):
+    '''Accept POST for only the delegate of the committee'''
+
+    def has_permission(self, request, view):
+        user = request.user
+        if user.is_superuser:
+            return True
+
+        method = request.method
+        committee_id = view.kwargs.get('committee', -1)
+        if method == 'POST' and user.is_authenticated() and user.is_delegate(
+        ) and user.delegate.assignment:
+            if user.delegate.committee_feedback_submitted:
+                raise ValidationError(
+                    {'detail': "Delegate feedback was already submitted."})
```

Comment:
I think PermissionDenied would be more appropriate here than a validation error

Reply:
Shouldn't PermissionDenied be related to permissions issues? This error is raised when a delegate has already submitted feedback and attempts to submit another which doesn't seem like a permissions issue. I also see how it's not really a validation issue either though because it has nothing to do with the data being passed in.

The only reason I put this in the permissions class at all is because I couldn't figure out a way access user.delegate data in the serializer when we could put a validator like in the school serializer. Is there a better way to do this or should be just leave it as PermissionDenied and call it there?

Reply:
I think PermissionDenied makes sense here; a delegate isn't allowed to submit more than once. Validation errors are typically for when there's an issue with a piece of the information you're trying to save, not really when you shouldn't be saving at all.

Reply:
Okay sounds good to me